### ISSUE TITLE
Inconsistent behavior between 'changed' and 'changing' signals for color picker

### ISSUE THREAD

**Issue by ntjess** [pyqtgraph/pyqtgraph#1385]
### Short description

I expect the returned value for `sigChanging` and `sigChanged` signals to be the same. E.g. if a text box is used, both signals should emit strings. In the case of a color picker, `sigChanged` emits a color (as expected) while `sigChanging` emits the color square that was picked. Either both should emit a color or both should emit the selected square, I think.

### Code to reproduce

```python
from pyqtgraph import parametertree
from pyqtgraph import mkQApp

app = mkQApp()
p = parametertree.Parameter.create(name='test', type='color', value='fff')

t = parametertree.ParameterTree()
t.setParameters(p, showTop=True)

p.sigValueChanging.connect(lambda _p, val: print(type(val)))
p.sigValueChanged.connect(lambda _p, val: print(type(val)))

t.show()
app.exec_()
```

### Expected behavior
Both types should match

### Real behavior
```
sigChanging vs. sigChanged:
<class 'pyqtgraph.widgets.ColorButton.ColorButton'>
<class 'PySide2.QtGui.QColor'>
```
### Tested environment(s)

 * PyQtGraph version: 0.11.0
 * Qt Python binding: 5.15
 * Python version: 3.8
 * Operating system: win10
 * Installation method: pip

**Comment by ixjlyons** [comment]
I believe the intention is to emit the value in both cases, so both should emit the `QColor` as far as I can tell. There's a comment pointing out the potential issue in the slot connected to `ColorButton.sigColorChanging`:

https://github.com/pyqtgraph/pyqtgraph/blob/b0a7c4435705747afe5fe5bf5c59c88e3609abd8/pyqtgraph/parametertree/parameterTypes.py#L209-L216

Indeed, the `ColorButton` widget emits itself only in its `sigColorChanging` signal.

So I guess we need to go through each of the types in `WidgetParameterItem` to see where this assumption is broken. I'm not sure why `widgetValueChanging` wouldn't just query the value with `self.widget.value()` like `widgetValueChanged` does - that seems like a much more robust way to do it, but that makes me think there's some reason it's not implemented that way.

That said, I'm not sure if the right fix is to fix `ColorButton` to emit both itself and its value in `sigColorChanging` (making it consistent with `SpinBox` for example) or to replace `args[-1]` with `self.widget.value()`.

Note that `GradientWidget` emits only its `GradientEditorItem` in both signals, so it has the same issue.

**Comment by lidstrom83** [comment]
By emitting the `GradientEditorItem` instead of a `ColorMap` value, the user has several more query options: `colorMap`, `getGradient`, `getColor`, `getLookupTable`. That said, the `ColorMap` object also has `map`, `getGradient`, `getColors`, `getLookupTable`, but HSV interpolation doesn't appear to be actually implemented here.

I wonder if the intention here was to implement HSV interpolation for `ColorMap` and then deprecate `GradientEditorItem.getGradient`, `GradientEditorItem.getColor`, `GradientEditorItem.getLookupTable`, or at least replace their internals with the equivalent calls to `ColorMap`.

I also can't tell the reason values are not emitted with these widgets' value changing signals. I suspect the reason is the same as the reason for not querying `self.widget.value()` in `widgetValueChanging`, namely that then we'd be consuming resources to rapidly create `QColor` or `ColorMap` objects even when they aren't being used.

I'm inclined to call this a bug as well because `Parameter.sigValueChanging` advertises that a value is emitted, and there should be consistency across types. I'll push the simple fix for this into #1469, and we can always change the widgets' behaviour later once it becomes clear how to proceed.
